### PR TITLE
reset dynamo in test_do_not_skip_side_effects unit test loop to avoid dynamo cache limit hit

### PR DESCRIPTION
**Pull request by laithsakka**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #128026
* #128023
* __->__ #127487

fix https://github.com/pytorch/pytorch/issues/127483

When nn module inlining is enabled, all recompilations are considered for the same frame hence we hit the cache limit for
test_do_not_skip_side_effects, but without inlining things are different , each time we hit a new Object Model we do not consider that a re-compilation, as explained in https://github.com/pytorch/pytorch/issues/127483

For that test we do not really care about cache size hence i reset dynamo in the main loop.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

### CODE REVIEW

File: test/dynamo/test_skip_non_tensor.py [hunk 1]
```diff
@@ -170,6 +170,8 @@ def test_do_not_skip_side_effects(self):
         global _variable, _variable_2
 
         for mode in range(1, 7):
+            torch._dynamo.reset()
```
Comment (anijain2305) [inline test/dynamo/test_skip_non_tensor.py:173]: What is the reason for recompilation?

Comment (anijain2305) [reply test/dynamo/test_skip_non_tensor.py:173]: Use TORCH_LOGS="recompiles"

Comment (laithsakka) [reply test/dynamo/test_skip_non_tensor.py:173]: https://github.com/pytorch/pytorch/issues/127483 Have all the details each iteration we _variable is updated and mode.

**Comment by laithsakka** [comment]
In general this raise a question if we want to convert nn module interger attributes  to symints when seen many times .

**Review by anijain2305 — APPROVED** [review]

**Pull request closed without merging**